pallets/pp/src/lib.rs

```diff
@@ -121,14 +223,28 @@ where
         caller: &T::AccountId,
         dest: &T::AccountId,
         gas_limit: Weight,
+        utxo_hash: H256,
+        utxo_value: u128,
+        fund_contract: bool,
         input_data: &Vec<u8>,
     ) -> Result<(), &'static str> {
-        let value = pallet_contracts::Pallet::<T>::subsistence_threshold();
+        // check if `dest` exist and if it does, update its balance information
+        <ContractBalances<T>>::get(&dest).ok_or("Contract doesn't exist!")?;
+        <ContractBalances<T>>::mutate(dest, |info| {
+            info.as_mut().unwrap().utxos.push((utxo_hash, utxo_value));
```

Comment:
Do we know this `unwrap()` is guaranteed to succeed? If so, a comment explaining why would be nice

Reply:
@altonen let's try like this

```rust
        <ContractBalances<T>>::mutate(dest, |info| {
            info.as_mut().map(|cb| cb.utxos.push((utxo_hash, utxo_value)));
        });
```

Reply:
I think this `unwrap()` is safe because the code above it (`get()`) checks that an entry for the smart contract exists and there's only one insert operation in the entire file that inserts a valid `Option` to the map. I added a comment explaining this